plugin-server/src/worker/ingestion/action-matcher.ts

```diff
@@ -137,16 +138,55 @@ export class ActionMatcher {
         this.statsd = statsd
     }
 
-    /** Get all actions matched to the event. */
     public async match(event: PostIngestionEvent, elements?: Element[]): Promise<Action[]> {
+        const legacyResponse = await this.matchLegacy(event, elements)
+        const hogVMResponse = this.matchBytecode(event, elements)
+
+        if (
+            legacyResponse.length !== hogVMResponse.length ||
+            !legacyResponse.every((action, index) => action.id == hogVMResponse[index]?.id)
+        ) {
+            this.statsd?.increment('action_matches_hogvm_legacy_mismatch')
+        } else {
+            this.statsd?.increment('action_matches_hogvm_legacy_match')
```

Comment:
This is a case for tags, which would make it easier to break down by the possible cases. We could have a single metric like `this.statsd?.increment('action_matching_completed', { hogvm_delta: Math.sign(hogVMResponse.length - legacyResponse.length) })` (1 means false positive, -1 means false negative, 0 means matching results)